disco/models/upgrade_cost_analysis_generic_input_model.py

```diff
@@ -0,0 +1,1166 @@
+
+class MiscUnitCostModel(UpgradeParamsBaseModel):
+    """Contains Miscellaneous Cost Database Model"""
+    
+    Description: str = Field(
+        title="Description",
+        description="Description of whether this is a fixed cost to add or replace transformer. "
+        "These are optional, and will be used if provided.",
+    )
+    total_cost: float = Field(
+        title="total_cost",
+        description="total_cost",
+    )
+    cost_units: str = Field(
+        title="cost_units",
+        description="Unit for cost. This should be in USD/unit",
+    )
+    
+    @validator("cost_units")
+    def check_misc_cost_units(cls, cost_units):
+        if cost_units not in ("USD/unit"):
+            raise ValueError("Incorrect cost units")
+        return cost_units
+    
+    @validator("Description")
+    def check_misc_description(cls, Description):
+        if Description not in ("Replace transformer (fixed cost)", "Add new transformer (fixed cost)"):
+            raise ValueError("Incorrect Miscellaneous Description")
+        return Description
+    
+
+class UpgradeCostDatabaseModel(UpgradeParamsBaseModel):
+    """Contains Upgrades Unit Cost Database needed for cost analysis"""
+    
+    transformers: List[TransformerUnitCostModel] = Field(
+        title="transformers",
+        description="This consists of all transformer unit costs",
+        default=[]
+    )
+    lines: List[LineUnitCostModel] = Field(
+        title="lines",
+        description="This consists of all line unit costs",
+        default=[]
+    )
+    control_changes: List[ControlUnitCostModel] = Field(
+        title="control_changes",
+        description="This consists of all control changes unit costs",
+        default=[]
+    )
+    voltage_regulators: List[VRegUnitCostModel] = Field(
+        title="voltage_regulators",
+        description="This consists of all voltage regulator unit costs",
+        default=[]
+    )
+    misc: List[MiscUnitCostModel] = Field(
+        title="misc",
+        description="This consists of all miscellaneous unit costs",
+        default=[]
+    )
+
+
+class upgrade_cost_types(enum.Enum):
+    """Possible values for upgrade costs"""
+    TRANSFORMER = "Transformer"
+    LINE = "Line"
+
+
+class CommonLineParameters(UpgradeParamsBaseModel):
+    """This model contains common line parameters that are used in linecode technical catalog, line technical catalog, line output upgrades. 
+    All these fields are directly available from opendss"""
+    r1: Any = Field(
+        title="r1",
+        description="r1",
+        upgrade_option_property=True,
+    )
+    x1: Any = Field(
+        title="x1",
+        description="x1",
+        upgrade_option_property=True,
+    )
+    r0: Any = Field(
+        title="r0",
+        description="r0",
+        upgrade_option_property=True,
+    )
+    x0: Any = Field(
+        title="x0",
+        description="x0",
+        upgrade_option_property=True,
+    )
+    C1: Any = Field(
```

Comment:
It's more predictable and straightforward if you always use lower-case, snake_case. There is quite a bit of inconsistency in this file.